timesketch/api/v1/resources/event.py

```diff
@@ -1208,3 +1248,159 @@ def post(self, sketch_id):
+            501,  # HTTP_STATUS_CODE_NOT_IMPLEMENTED
+            ("Method not yet implemented. Please use EventTaggingResource instead."),
+        )
+
+    @login_required
+    def delete(self, sketch_id):
+        """
+        Remove tags (max 500) from a list of events (max 500).
+
+        Args:
+            sketch_id: Integer primary key for a sketch database model
+            in request form:
+                events: list of events to remove tags from with the following values:
+                    _id: the event id
+                    _index: the searchindex id or
+                    _searchindex_name: the searchindex name
+                tags: list of tags to remove from events
+
+
+        Returns:
+            HTTP_STATUS_CODE_OK if successful
+
+        Raises:
+            HTTP_STATUS_CODE_NOT_FOUND: if sketch or event does not exist
+            HTTP_STATUS_CODE_BAD_REQUEST: if the request is malformed, e.g.
+                events or tags are not lists or the number of events or tags
+                exceeds the maximum allowed
+            HTTP_STATUS_CODE_FORBIDDEN: if the user does not have write access
+                to the sketch
+        """
+
+        sketch = Sketch.query.get_with_acl(sketch_id)
+        if not sketch:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No sketch found with this ID.")
+
+        if not sketch.has_permission(current_user, "write"):
+            abort(
+                HTTP_STATUS_CODE_FORBIDDEN,
+                "User does not have write access controls on sketch.",
+            )
+
+        form = request.json
+        if not form:
+            form = request.data
+
+        events = form.get("events", [])
+        if not isinstance(events, list):
+            abort(HTTP_STATUS_CODE_BAD_REQUEST, "Events need to be a list")
+
+        if len(events) > self.MAX_EVENTS_TO_TAG:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Cannot untag more than {0:d} events in a single "
+                "request".format(self.MAX_EVENTS_TO_TAG),
+            )
+
+        tags = form.get("tags", [])
+        if not isinstance(tags, list):
+            abort(HTTP_STATUS_CODE_BAD_REQUEST, "Tags need to be a list")
+
+        if len(tags) > self.MAX_TAGS_PER_REQUEST:
+            abort(
+                HTTP_STATUS_CODE_BAD_REQUEST,
+                "Cannot untag more than {0:d} tags in a single "
+                "request".format(self.MAX_TAGS_PER_REQUEST),
+            )
+
+        datastore = self.datastore
+
+        for _event in events:
+            # every event entry can have a dedicated searchindex_id or searchindex_name
+            searchindex_id = _event.get("_searchindex_id")
+            searchindex_name = _event.get("_searchindex_name")
+
+            if not (searchindex_id or searchindex_name):
+                abort(
+                    HTTP_STATUS_CODE_NOT_FOUND,
+                    "No search index information supplied.",
+                )
```

Comment:
You could probably remove this check, since all events should have an `_index` field.

Reply:
not if one was building it manually

Reply:
All events will have a _index field, regardless of where they were imported from. `_index` is a special opensearch field that is added by the system.

Reply:
yes true, but this is about potentially user supplied info, it is true that in most of our cases, this will be executed after one pulled an event. That does not rule out that someone is using the method just providing e.g. an event id but not an index.